quickstart/ia32-generic-qemu.md

```diff
@@ -124,6 +124,141 @@ The number of detected cores is presented during kernel initialization.
+      systemctl start libvirtd.service
+      ```
+
+    - Start the default network bridge, and configure it to run on startup.
+
+      ```bash
+      sudo virsh net-autostart --network default
+      sudo virsh net-start --network default
+      ```
+
+    - If IPv6 is needed change the configuration of `virbr0`
+
+      ```bash
+      sudo virsh net-destroy default
+      sudo virsh net-edit default
+      ```
+
+      The commands above open the editor of the configuration file of `virbr0`. The are two necessary changes:
+
+      - Add IPv6 address to the bridge interface:
+
+        ```xml
+        <ip family='ipv6' address='2001:db8:dead:beef:fe::2' prefix='64'>
+        </ip>
+        ```
+
+      - Enable NAT for IPv6:
+
+        ```xml
+        <forward mode='nat'>
+          <nat ipv6='yes'/>
+        </forward>
+        ```
+
+      The overall config should look something like this:
+
+      ```xml
+      <network ipv6='yes'>
+        <name>default</name>
+        <uuid>a9e032b7-e32f-4f91-a273-e6c6f15b8904</uuid>
+        <forward mode='nat'>
+          <nat ipv6='yes'/>
+        </forward>
+        <bridge name='virbr0' stp='on' delay='0'>
+        <mac address='52:54:00:99:4d:c3'/>
+        <ip address='192.168.122.1' netmask='255.255.255.0'>
+          <dhcp>
+            <range start='192.168.122.2' end='192.168.122.254'/>
+          </dhcp>
+        </ip>
+        <ip family='ipv6' address='2001:db8:dead:beef:fe::2' prefix='64'>
+        </ip>
+      </network>
+      ```
+
+      Save the config file and start the bridge by running:
+
+      ```bash
+      sudo virsh net-start default
+      ```
+
+    - After that verify that the IP range `192.168.122.1/24` is reported by the `vibr0` bridge:
+
+      ```bash
+      ip addr show virbr0
+      ```
+
+    - The expected output:
+
+      ```bash
+       virbr0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue state DOWN group default qlen 1000
+          link/ether xx:xx:xx:xx:xx:xx brd ff:ff:ff:ff:ff:ff
+          inet 192.168.122.1/24 brd 192.168.122.255 scope global virbr0
+             valid_lft forever preferred_lft forever
+      ```
+
+    - Set up `qemu-bridge-helper` (`chmod` is used here to allow running QEMU without root privileges)
+
+      ```bash
+      echo "allow virbr0" > /etc/qemu/bridge.conf
+      sudo chmod a+rw /etc/qemu/bridge.conf
+      ```
+
+    - If `/etc/qemu` directory does not exist, create it and provide required privileges:
+
+      ```bash
+      sudo mkdir /etc/qemu
+      sudo chmod a+rw /etc/qemu
+      echo "allow virbr0" > /etc/qemu/bridge.conf
+      sudo chmod a+rw /etc/qemu/bridge.conf
+      ```
+
+      ![Image](_images/ia32_sdk_vibr_setup.png)
+
+    - Sources: <https://apiraino.github.io/qemu-bridge-networking/>,
+     <https://mike42.me/blog/2019-08-how-to-use-the-qemu-bridge-helper-on-debian-10>
+
+ 2. Launch `qemu` using a starting script with `net` suffix:
+
+      ```bash
+      ./scripts/ia32-generic-qemu-net.sh
```

Comment:
I have:
```
failed to create tun device: Operation not permitted
qemu-system-i386: -netdev bridge,br=virbr0,id=net0: bridge helper failed
```

Reply:
I run it with sudo, but will try to find another way